internal/address/xrp.go

```diff
@@ -0,0 +1,73 @@
+package address
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	"golang.org/x/crypto/ripemd160"
+)
+
+// Base58 alphabet used by XRP
+const alphabet = "rpshnaf39wBUDNEGHJKLM4PQRST7VWXYZ2bcdeCg65jkm8oFqi1tuvAxyz"
+
+// AccountID prefix for XRP addresses
+var accountIDPrefix = []byte{0x00}
+
+func GetXRPAddress(hexPublicKey string) (string, error) {
+	publicKey, err := hex.DecodeString(hexPublicKey)
+	if err != nil {
+		return "", fmt.Errorf("invalid hex public key: %v", err)
+	}
+	sha := sha256.New()
+	sha.Write(publicKey)
+	hash := sha.Sum(nil)
+
+	ripemd := ripemd160.New()
+	ripemd.Write(hash)
+	hash = ripemd.Sum(nil)
+
+	versionHash := append([]byte{0}, hash...)
+
+	sha = sha256.New()
+	sha.Write(versionHash)
+	hash = sha.Sum(nil)
+
+	sha = sha256.New()
+	sha.Write(hash)
+	hash = sha.Sum(nil)
+
+	checksum := hash[:4]
+
+	finalHash := append(versionHash, checksum...)
+	return base58Encode(finalHash), nil
+}
+
+func base58Encode(input []byte) string {
```

Comment:
there is a method in `"github.com/cosmos/btcutil/base58"`, so we don't need to create our own

Reply:
Fixed